librz/include/rz_core.h

```diff
@@ -770,6 +771,22 @@ RZ_API char *rz_core_sysenv_begin(RzCore *core, const char *cmd);
 RZ_API void rz_core_sysenv_end(RzCore *core, const char *cmd);
 
 RZ_API void rz_core_recover_vars(RzCore *core, RzAnalysisFunction *fcn, bool argonly);
+
+/* linux_heap_glibc.c */
```

Comment:
The fact that the malloc implementations have to stay in RzCore seems a bit wrong, but definitely not something to consider in this PR.

However, do we really need these API exported  in the public API? Can we put them as RZ_IPI in core_private.h? If they have to stay in the public API they should be at least prefixed with `rz_core_`.

Reply:
I agree that ideally they should be moved to `librz/debug`, except configuration handling and command handlers. But, indeed, this is out of the scope of this PR.

Reply:
Also agree on keeping these APIs private.

Reply:
@PulakIIIT don't forget about this one.

Reply:
@XVilka @ret2libc I added them as public APIs as in the future I will need some of the malloc-related APIs when I implement the heap viewer in Cutter

Reply:
Then I think you need better names for them. Something like `rz_debug_heap_` or `rz_heap_`. Currently it is quite inconsistent.

Reply:
At the moment I have renamed the functions to be more consistent, in the following weeks I will be working on the heap viewer in Cutter so I will keep updating the Rizin API for heaps as I need and I will remove any API calls which are not required.